### ISSUE TITLE
`composer show -s -t --no-ansi` prints error about undefined variable

### ISSUE THREAD

**Issue by Wirone** [composer/composer#6902]
My `composer.json` is not related since it occurs in every repo when using Composer v1.5.5 (2017-12-01 14:42:57). My colleague checked on v1.5.2 and it's working fine.

Output of `composer diagnose`:

```
Checking composer.json: WARNING
Defining autoload.psr-4 with an empty namespace prefix is a bad idea for performance
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer show -s -t --no-ansi
```

I get the following output:

```
acme/some-package master Acme Package
|--php >=7.0
|--ext-intl *
|--symfony/symfony ~3.0
|--symfony/assetic-bundle ~2.7
|--symfony/swiftmailer-bundle ~2.3
|--symfony/monolog-bundle ~3.0
|--sensio/distribution-bundle ~5.0
|--sensio/framework-extra-bundle ~3.0,>=3.0.11
|--doctrine/orm ~2.5
|--doctrine/doctrine-bundle ~1.5,>=1.5.1
|--doctrine/doctrine-fixtures-bundle ~2.3
|--doctrine/data-fixtures ~1.1,<1.2
|--doctrine/doctrine-migrations-bundle ~1.1
|--twig/extensions ~1.0
|--incenteev/composer-parameter-handler ~2.0
|--guzzlehttp/guzzle ~5.3
|--friendsofsymfony/jsrouting-bundle ~1.6
|--friendsofsymfony/rest-bundle ~1.6
|--nelmio/api-doc-bundle ~2.7
|--jms/job-queue-bundle ~1.1
|--jms/serializer-bundle ~1.0
|--pagerfanta/pagerfanta ~1.0
|--phpoffice/phpexcel ~1.8
`--graylog2/gelf-php ~1.0

                                
  [ErrorException]              
  Undefined variable: exitCode  
                                

show [--all] [-i|--installed] [-p|--platform] [-a|--available] [-s|--self] [-N|--name-only] [-P|--path] [-t|--tree] [-l|--latest] [-o|--outdated] [-m|--minor-only] [-D|--direct] [--strict] [-f|--format FORMAT] [--] [<package>] [<version>]
```

And I expected this to happen: no errors displayed.

**Comment by stof** [comment]
can you run it in verbose mode (with `-v` to get the stack trace ?

**Comment by Wirone** [comment]
Sure, here you are:

```
[ErrorException]              
  Undefined variable: exitCode  
                                

Exception trace:
 () at phar:///usr/local/bin/composer/src/Composer/Command/ShowCommand.php:225
 Composer\Util\ErrorHandler::handle() at phar:///usr/local/bin/composer/src/Composer/Command/ShowCommand.php:225
 Composer\Command\ShowCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:266
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:861
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:208
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:245
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:127
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:100
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:54
 require() at /usr/local/bin/composer:24
```

**Comment by tomzx** [comment]
@Seldaek This can be closed because of #6907.

**Comment by Seldaek** [comment]
Ah sorry I spotted #6882 was also fixed by this and I didn't check which issue you referenced, just assumed that was it :s Anyway sure closing this then thanks.
